zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -1316,6 +1316,44 @@ public void shouldPropagateVariablesOfCompensationHandler() {
             tuple(processInstanceKey, "global", "2"));
   }
 
+  @Test
+  public void shouldTriggerCompensationForMultiInstanceActivityOnlyOnce() {
+    final var process =
+        createModelFromClasspathResource("/compensation/compensation-multi-instance-activity.bpmn");
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withType(SERVICE_TASK_TYPE_COMPENSABLE_ACTIVITY)
+        .limit(3)
+        .forEach(job -> ENGINE.job().withKey(job.getKey()).complete());
```

Comment:
:wrench: We have now a small helper for completing jobs.

```suggestion
    completeJobs(processInstanceKey, SERVICE_TASK_TYPE_COMPENSABLE_ACTIVITY, 3);
```